### DISCUSSION TITLE
How to correctly set sidekiq, sidekiq-pro and sidekiq-ent version in Gemfile?

### DISCUSSION THREAD

**Discussion by Fire-Dragon-DoL** [sidekiq/sidekiq#6168] [Q&A] [answered]
I read the [documentation in the wiki](https://github.com/sidekiq/sidekiq/wiki/Commercial-FAQ#how-do-i-upgrade-sidekiq-pro-or-sidekiq-enterprise) on the topic, but I found myself in a situation where sidekiq was version 7.2.0, but sidekiq-pro and sidekiq-ent were set to version 7.0.0. This is a known incompatible state.

I would like to avoid this situation in the future, what's the correct approach?

My idea was to set the version of **only sidekiq-ent** and let the other two "tag along":

```ruby
gem "sidekiq"
source "https://enterprise.contribsys.com/" do
  gem "sidekiq-pro"

  gem "sidekiq-ent", "~> 7.2"
end
```

The documentation disagrees with me, I'd love to hear the rationale for following the documentation vs the approach I recommended.

**Comment by mperham** [comment, marked as answer]
I recommend locking Sidekiq and letting the commercial gems float. You need a stable foundation:

gem "sidekiq", "~> 7.2.0"

will lock to 7.2.x. Remember you upgrade with `bundle up sidekiq-ent`, **not** bundle up sidekiq.
